StRoot/StRHICfUtil/StRHICfFunction.h

```diff
@@ -0,0 +1,73 @@
+/***************************************************************************
+ * Author: Seunghwan Lee
+ ***************************************************************************
+ *
+ * Description: RHICf Useful class for calibration and reconstruction
+ *
+ ***************************************************************************
+ * $Id: StRHICfFunction.h,v 1.0 2022/08/25 2:51:00 SeunghwanLee Exp $
+ * $Log: StRHICfFunction.h,v $
+ ***************************************************************************/
+
+#ifndef StRHICfFunction_H
+#define StRHICfFunction_H
+
+#include <algorithm>
+#include <iostream>
+#include <string>
+#include "StRoot/StEvent/StEnumerations.h"
+
+using namespace std;
+
+class StEnumerations;
```

Comment:
Is this line needed if you have above:

    #include "StRoot/StEvent/StEnumerations.h"

?

Reply:
Hello Gene, 

There are some functions that check whether there is correct input data or not. (this is just checking the size of the data)
So some parameters like kRHICfNtower, kRHICfNlayer, and so on are in this source file.

If someone uses the RHICf reconstruction tool without changing something, these checking functions will be meaningless.

If you think not needed these checking functions, then I will remove these functions and StEnumerations.hh include line.

Thank you

Reply:
Thanks for your reply, but I think I was not clear. I am saying that _if_ you have the `#include` statement, you _do_not_need_ the `class StEnumerations;` statement because the StEnumerations class is already known from the `#include` statement. You can delete the `class StEnumerations;` line.

Reply:
And there is more...

I do not see the use of any functionality defined in `StEnumerations.h` used here in `StRHICfFunctions.h`. So one question to ask is "why have the `#include` for `StEnumerations.h` inside `StRHICfFunctions.h`"?

You should answer this question.

One possible answer is that every other code that includes `StRHICfFunctions.h` also needs functions defined in `StEnumerations.h`, so this is a shortcut to avoid having to include `StEnumerations.h` in those other codes.

Another possible answer is that it may be better to change the codes to include `StEnumerations.h` only in those other codes that really need it, and remove both the `class` and the `#include` for `StEnumerations` from `StRHICfFunctions.h`.

I hope that is clear, but please let me know if I need to explain the question better.

Thanks,
-Gene

Reply:
class StEnumerations; this line is my mistake. sorry that is ok if removed.
I will remove it.

and I mean, some constant values for RHICf in the **StEnumerations.h** are used in the **StRHICfFunctions.cxx** source file.
In the **StRHICfFunctions.cxx**, you can see some values like a kRHICfNtower, kRHICfNlayer, etc.. 

the **StRHICfFunction.h** class is called in other classes such as **StRHICfHitMaker**, and so on.
so as you say, the **StEnumeration** class should be better to call only one time using the **StRHICfFunction** in the RHICf library
Likewise, I will remove including part for **StEnumeration.h** in other classes that already inherited **StRHICfFunction** 

Thank you for your comments.

Reply:
I had committed the code. Thank you.